contents/docs/contribute/developing-locally.mdx

```diff
@@ -9,231 +9,268 @@ showTitle: true
     special handling related to releases.
 </blockquote>
 
-<HiddenSection headingType='h2' title='Using virtualenv' defaultIsOpen>
+## What does it take?
 
-> **Note:** If you're using the latest generation MacBooks (M1) have a look [at this issue](https://github.com/PostHog/posthog/issues/2916) for info on how to run PostHog.
+To run PostHog, you must run the following services:
 
-> If you want to run your development instance using **ClickHouse**, check out our [EE setup](/handbook/engineering/ee-setup) section (currently intended for internal use only).
+- PostHog's Django Backend
+- PostHog's Celery Worker
+- PostHog's NodeJS Plugin Server
+- PostHog's NodeJS Frontend
+- PostgreSQL
+- Redis
+- ClickHouse
+- Kafka
+- Zookeeper
 
-### Clone PostHog
+The guide below explains how to get them all running.
 
-1. Clone the repository:
+We officially support only one way of running PostHog, where all services (Postgres, Redis, ClickHouse, Kafka, Zookeeper) run via Docker, with PostHog's apps running natively on your system.
+For a while, we used to support running the PostHog apps themselves in Docker, but recently deprecated this approach because 1) that made everything really really slow, 2) you need to install nodejs and python locally anyway if you want to commit any code due to our pre-commit hooks (black & prettier). Thus to simplify instructions, we consolidated on the steps below.
+
+The instructions below assume you're running MacOS. For Linux, adjust the steps as needed (e.g. use `apt` instead of `brew`).
+
+In case some steps here have fallen out of date, please tell us about it by [submitting a patch](https://github.com/PostHog/posthog.com/tree/master/contents/docs/contribute/developing-locally.mdx).
+
+## Mac OS prerequisites
+
+1. Install xcode developer tools if you haven't already: `xcode-select --install`.
+
+2. Install Homebrew by following the [instructions here](https://brew.sh/).
+
+<blockquote class="warning-note">
+    Note: After installation, make sure to follow the instructions printed in your terminal to add homebrew to your path. Same for <code>nvm</code> and <code>pyenv</code> below.
+</blockquote>
+
+3. Install [Docker Desktop](https://www.docker.com/products/docker-desktop) and give it **at least 4 GB of RAM** (6GB if you can afford it) and 8 CPU cores.
+
+4. Add the following lines to `/etc/hosts`:
```

Comment:
I have some context:

I don't think this is needed and will cause trouble for people running multiple projects at once.

Instead I have these following env vars set up for myself

```
⟩ export KAFKA_ENABLED=true KAFKA_HOSTS=localhost:9092 KAFKA_URL=kafka://localhost:9092
```

Frustratingly, there's two url ones because plugin-server and app use different ones.

There's a gotcha here around `migrate_clickhouse` though - since clickhouse accesses network over docker, the host needs to be `kafka` there but `localhost` locally. I think we could improve this by better defaults in the app though.

Reply:
> I don't think this is needed 

Sorry, but things just don't work for me without this... so I do think it's needed. ClickHouse and Kafka are unable to communicate. Probably the same for other Docker on Mac users (please confirm!). I'll try once more without, but happy to add a "skip if non mac" line there.

Reply:
I think the reason it didn't work it's because of the `KAFKA_HOSTS` setting being set when `python manage.py migrate_clickhouse` is run.

If we go the route of "sane defaults if DEBUG=1 is set" I think I have a solution for that and can solve this in a follow-up. This would be conditionally setting `KAFKA_HOSTS` to kafka if DEBUG during `migrate_clickhouse` so the kafka engine table is set up correctly :)